lighthouse-core/lib/page-functions.js

```diff
@@ -114,7 +113,35 @@ function getElementsInDocument(selector) {
 function getOuterHTMLSnippet(element) {
   const reOpeningTag = /^.*?>/;
   const match = element.outerHTML.match(reOpeningTag);
-  return match && match[0] || '';
+  return (match && match[0]) || '';
+}
+
+/**
+ * Computes a memory/CPU performance benchmark index to determine rough device class.
+ * @see https://docs.google.com/spreadsheets/d/1E0gZwKsxegudkjJl8Fki_sOwHKpqgXwt8aBAfuUaB8A/edit?usp=sharing
+ *
+ * The benchmark creates a string of length 100,000 in a loop.
+ * The returned index is the number of numbers per second the string can be created.
+ *
+ *  - 750+ is a desktop-class device, iPhone X, etc
+ *  - 300+ is a high-end Android phone, Galaxy S8, low-end Chromebook, etc
+ *  - 75+ is a mid-tier Android phone, Nexus 5X, etc
+ *  - <75 is a budget Android phone, Alcatel Ideal, Galaxy J2, etc
+ */
+/* istanbul ignore next */
+function ultradumbBenchmark() {
```

Comment:
I know we just had a whole thing with page-functions, but I kind of wish this was in its own file for easier running :) I guess we can always add `yarn ultradumbBenchmark` later

Reply:
done :)
![image](https://user-images.githubusercontent.com/2301202/44476991-45675700-a5ee-11e8-964f-f13dea20668d.png)